### PR TITLE
Feat 108 Add pickle option

### LINKED ISSUE #108: Add option to return pickled data instead of json
# User story
As a user, I want to de-serialize pickled data, so I can easily extract the pydantic models if needed.

# Acceptance criteria
- [ ] Add an option in the get request so that the response returned to the user is an octet-stream of pickled data.

# Sprint Ready Checklist 
- [ ] 1. Acceptance criteria defined
- [ ] 2. Team understands acceptance criteria
- [ ] 3. Team has defined solution / steps to satisfy acceptance criteria
- [ ] 4. Acceptance criteria is verifiable / testable
- [ ] 5. External / 3rd Party dependencies identified
- [ ] 6. Ticket is prioritized and sized

# Notes
*Add any helpful notes here.*

### PR DESCRIPTION
**Pull request by Louise-X10**
close #108

### CODE REVIEW

File: src/aind_metadata_service/server.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 
 @app.get("/subject/{subject_id}")
-async def retrieve_subject(subject_id):
+async def retrieve_subject(subject_id, pickle=False):
```
Comment (jtyoung84) [inline src/aind_metadata_service/server.py:42]: Maybe add a comment or a note in the docstring that the pickle parameter will be handled in a future version

File: tests/test_client.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ class TestAindMetadataServiceClient(unittest.TestCase):
 
     @mock.patch("requests.get")
     def test_get_subject(self, mock_get: MagicMock) -> None:
-        """Tests the get_subject method."""
+        """Tests the get_subject method (with default pickle = False)."""
```
Comment (jtyoung84) [inline tests/test_client.py:17]: Add a test where pickle is set to true

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
See comments

**Comment by Louise-X10** [comment]
Separate ticket for implementing conversion to pickled data (#120)

File: src/aind_metadata_service/server.py (hunk 1 above)
Comment (jtyoung84) [reply src/aind_metadata_service/server.py:42]: Also add a note here in the server.py file

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
Also add a note in the server.py file

**Review by jtyoung84 — APPROVED** [review]

**Pull request merged**